modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/api/ApiAppHandler.java

```diff
@@ -37,7 +38,8 @@
 public class ApiAppHandler
     extends BaseWebHandler
 {
-    public static final Pattern PATTERN = Pattern.compile( "^/api/(?<appKey>(?!media|idprovider)[^/]+)/(?<restPath>.*)?$" );
+    public static final Pattern PATTERN =
+        Pattern.compile( "^/api/(?<appKey>(?!media|idprovider)[^/]+)/(?<apiKey>[^/]+)/(?<restPath>.*)?$" );
```

Comment:
idprovider is no longer in /api

Reply:
Yes, but that is within `(?!...)` in other word it must not match. I think that part will be extended the rest of reserved words.